src/lint/mod.rs

```diff
@@ -43,8 +49,9 @@ pub fn maybe_parse_lint_cfg(path: PathBuf) -> Option<LintCfg> {
 
 #[derive(Clone, Debug, Serialize, Deserialize, PartialEq)]
 #[serde(default)]
-#[serde(deny_unknown_fields)]
 pub struct LintCfg {
+    #[serde(flatten)]
+    pub unknown_fields: HashMap<String, Value>,
```

Comment:
No reason to store this in the struct. The function should return the extra information separately (either as a return tuple, or into a mut ref argument)

You can look at the try_deserialize function in config.rs for how this is done for the server-level configuration. It has support for duplicate, unknown, and deprecated options.